vendor/directx/d3d12/d3d12_constants.odin

```diff
@@ -498,6 +498,18 @@ WHQL_DRAW_VERTEX_COUNT_2_TO_EXP :: 25
 
 SHADER_COMPONENT_MAPPING_MASK :: 0x7
 SHADER_COMPONENT_MAPPING_SHIFT :: 3
+SHADER_COMPONENT_MAPPING_ALWAYS_SET_BIT_AVOIDING_ZEROMEM_MISTAKES :: 1<<(SHADER_COMPONENT_MAPPING_SHIFT*4)
+ENCODE_SHADER_4_COMPONENT_MAPPING :: proc(Src0, Src1, Src2, Src3: SHADER_COMPONENT_MAPPING) -> u32 {
+	return	(u32(Src0)&SHADER_COMPONENT_MAPPING_MASK)|
+			((u32(Src1)&SHADER_COMPONENT_MAPPING_MASK)<<SHADER_COMPONENT_MAPPING_SHIFT)|
+			((u32(Src2)&SHADER_COMPONENT_MAPPING_MASK)<<(SHADER_COMPONENT_MAPPING_SHIFT*2))|
+			((u32(Src3)&SHADER_COMPONENT_MAPPING_MASK)<<(SHADER_COMPONENT_MAPPING_SHIFT*3))|
+			SHADER_COMPONENT_MAPPING_ALWAYS_SET_BIT_AVOIDING_ZEROMEM_MISTAKES
+}
+DECODE_SHADER_4_COMPONENT_MAPPING :: proc(ComponentToExtract, Mapping: u32) -> SHADER_COMPONENT_MAPPING {
```

Comment:
I recommend making these `proc "c"` or something that is the default `"odin"` calling convention